cpop_bot/text_handler.py

```diff
@@ -36,16 +36,17 @@ async def handle(self, message: types.Message):
             await self.__delete_messages(DELAY_DELETE_IN_SEC_PING,
                                          (pong, message))
 
-    async def __is_provide_service(self, message: types.Message):
+    async def __should_provide_service(self, message: types.Message):
         chat_id = message.chat.id
         if chat_id in WHITELIST_CHAT_ID:
             return True
         chat_type = message.chat.type
         if chat_type == "private":
-            member = await self.bot.get_chat_member(GROUP_CHAT_ID, chat_id)
-            if member.is_chat_member():
-                return True
-            else:
+            user_id = chat_id
+            for promotion_chat in PROMOTION_LIST_CHAT_ID:
+                if (await self.bot.get_chat_member(
+                        promotion_chat, user_id)).is_chat_member():
+                    return True
                 await message.reply(AVAILABILITY_HTML,
                                     parse_mode=ParseMode.HTML)
                 return False
```

Comment:
looks like you can remove some returns. 

Right now your code looks like this
```
    async def __should_provide_service(self, message: types.Message):
        chat_id = message.chat.id
        if chat_id in WHITELIST_CHAT_ID:
            return True
        chat_type = message.chat.type
        if chat_type == "private":
            user_id = chat_id
            for promotion_chat in PROMOTION_LIST_CHAT_ID:
                if (await self.bot.get_chat_member(
                        promotion_chat, user_id)).is_chat_member():
                    return True
                await message.reply(AVAILABILITY_HTML,
                                    parse_mode=ParseMode.HTML)
                return False
        else:
            return False
```

You can change it to this:
```
    async def __should_provide_service(self, message: types.Message):
        chat_id = message.chat.id
        if chat_id in WHITELIST_CHAT_ID:
            return True
        chat_type = message.chat.type
        if chat_type == "private":
            user_id = chat_id
            for promotion_chat in PROMOTION_LIST_CHAT_ID:
                if (await self.bot.get_chat_member(
                        promotion_chat, user_id)).is_chat_member():
                    return True
                await message.reply(AVAILABILITY_HTML,
                                    parse_mode=ParseMode.HTML)
        return False
```

Reply:
That's a for loop inside if else statement, not just if inside if.

with your suggestion, if the user is not member of any chat in PROMOTION_LIST_CHAT_ID, I guess the the function will return None in the final loop.

with the current commit in the PR, the for loop returns "True" or "False", if it always return True or False, then we can remove the else on the first level. But I'm not sure if it always return True/False (let's assume that in some extreme case that the for loop fails before return?)

also pylint does not warn about that. there is no "no-else-return" warns for now.

Reply:
right, the for loop shouldn't return False but return False at the end of the function. new commit solved this.